src/libraries/System.Runtime.InteropServices/gen/LibraryImportGenerator/Analyzers/CustomMarshallerAttributeFixer.cs

```diff
@@ -93,42 +131,267 @@ public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+                    if (diagnostic.Properties.TryGetValue(MissingMemberNames.Key, out string missingMembers))
                     {
-                        missingMemberNames.AddRange(missingMembers.Split(CustomMarshallerAttributeAnalyzer.MissingMemberNames.Delimiter));
+                        missingMemberNames.UnionWith(missingMembers.Split(MissingMemberNames.Delimiter));
                     }
                 }
             }
 
-            return (missingMemberNames, requiredShapeDiagnostics);
+            return (new HashSet<string>(missingMemberNames), requiredShapeDiagnostics);
         }
 
-        private static async Task<Document> AddMissingMembers(Document doc, SyntaxNode node, List<string> missingMemberNames, CancellationToken ct)
+        private static async Task<Solution> AddMissingMembers(Document doc, SyntaxNode node, HashSet<string> missingMemberNames, CancellationToken ct)
         {
-            var editor = await DocumentEditor.CreateAsync(doc, ct).ConfigureAwait(false);
-            var gen = editor.Generator;
+            var model = await doc.GetSemanticModelAsync(ct).ConfigureAwait(false);
+
+            var entryPointTypeSymbol = (INamedTypeSymbol)model.GetEnclosingSymbol(node.SpanStart, ct);
+
+            // TODO: Convert to use the IOperation tree once IAttributeOperation is available
+            var managedTypeSymbolInAttribute = GetManagedTypeInAttributeSyntax(node.GetLocation(), entryPointTypeSymbol);
+
+            bool isLinearCollectionMarshaller = ManualTypeMarshallingHelper.IsLinearCollectionEntryPoint(entryPointTypeSymbol);
+
+            ManualTypeMarshallingHelper.TryResolveManagedType(entryPointTypeSymbol, ManualTypeMarshallingHelper.ReplaceGenericPlaceholderInType(managedTypeSymbolInAttribute, entryPointTypeSymbol, model.Compilation), isLinearCollectionMarshaller, (_, _) => { }, out ITypeSymbol managedType);
 
-            SyntaxNode updatedDeclaration = AddMissingMembers(node, (ITypeSymbol)editor.SemanticModel.GetDeclaredSymbol(node, ct), missingMemberNames, editor.SemanticModel.Compilation, gen);
+            SymbolEditor editor = SymbolEditor.Create(doc.Project.Solution);
 
-            editor.ReplaceNode(node, updatedDeclaration);
+            INamedTypeSymbol marshallerType = (INamedTypeSymbol)model.GetSymbolInfo(node, ct).Symbol;
 
-            return editor.GetChangedDocument();
+            await editor.EditOneDeclarationAsync(marshallerType, (editor, decl) => AddMissingMembers(editor, decl, marshallerType, managedType, missingMemberNames, isLinearCollectionMarshaller), ct).ConfigureAwait(false);
+
+            return editor.ChangedSolution;
+        }
+
+        private static ITypeSymbol? GetManagedTypeInAttributeSyntax(Location locationInAttribute, INamedTypeSymbol? attributedTypeSymbol)
+            => (ITypeSymbol)attributedTypeSymbol.GetAttributes().First(attr =>
```

Comment:
Isn't `attributedTypeSymbol` nullable? Was this checked somewhere?

Reply:
I need to figure out what's going on here in our infra. I feel like the nullable warnings have been suppressed for some reason. I've been getting weird stuff like this for a bit (where obvious places for warnings aren't showing up)

Reply:
I haven't had a chance to investigate what's going on with our infrastructure. In the meantime, I've updated the nullability annotations here based on usage to be more accurate and added a comment explaining why we don't need to worry about nulls here.